### PR TITLE
Create Kids Dialog

### LINKED ISSUE #2509: [Kids profile] Create the first screen
Design: G60UqxPhTynbalo0Vfr1NC-fi-1853_16704

### LINKED ISSUE #2510: [Kids profile] Create the second screen
Design: G60UqxPhTynbalo0Vfr1NC-fi-1853_17221

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- Creates the following forms for kids feature when tapping on `Request Early Access`
- See: G60UqxPhTynbalo0Vfr1NC-fi-1853_16704

I will work on the transition between the two new dialogs in a later PR 

Fixes #2510
Fixes #2509

## Testing Instructions
1. Have KIDS_PROFILE feature flag enabled
2. Go to profile
3. Tap on Request Early access
4. You should see the `Thank you for your interest` bottom sheet dialog
5. Tap on `No, thank you` button
6. The dialog should be dismissed
7. Tap on Request Early access again
8. Tap on `Send feedback` button
9. The next dialog with a form should be opened

## Screenshots or Screencast 
[Screen_recording_20240731_103005.webm](https://github.com/user-attachments/assets/0aeed71f-b267-440b-bc96-66e6360f5625)

image

image

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsBottomSheetDialog.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+@AndroidEntryPoint
+class KidsBottomSheetDialog : BottomSheetDialogFragment() {
+
+    @Inject
+    lateinit var theme: Theme
+
+    private val viewModel: KidsSendFeedbackViewModel by viewModels()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        val context = context ?: throw Exception("Context not found")
```
Comment (MiSikora) [inline modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsBottomSheetDialog.kt:31]: There is a `requireContext()` function available for fragments. That does exactly that.

File: modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsDialog.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                modifier = modifier.padding(bottom = 12.dp),
+            )
+
+            TextH50(
+                text = stringResource(LR.string.kids_bottom_sheet_description),
+                fontWeight = FontWeight.W500,
+                color = MaterialTheme.theme.colors.primaryText02,
+                textAlign = TextAlign.Center,
+                modifier = modifier.padding(bottom = 12.dp),
+            )
```
Comment (MiSikora) [inline modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsDialog.kt:79]: In the designs these two text fields have some horizontal padding to it so they are not aligned exactly with the buttons.

**Review by MiSikora — APPROVED** [review]
Nice work. I left some minor comments and have 3 more here.

1. When the text field is in focus keyboard covers the submit button

Screenshot_20240730-161232

2. It feels a bit awkward that you need to tap exactly on the `Request Early Access` text in order to start the flow. I think it would be better to start if a user taps anywhere on the banner (except the `x`).

3. Transition between first and second dialog feels a bit abrupt and could use a simple animation.

File: modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsDialog.kt (hunk 2 above)
Comment (mebarbosa) [reply modules/features/kids/src/main/java/au/com/shiftyjelly/pocketcasts/kids/KidsDialog.kt]: Nice catch! I've updated the code with this

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by mebarbosa** [comment]
Thank you @MiSikora for all feedbacks! I addressed here most of them, but I can address any other in another PR

**Pull request merged**